src/composeApolloProvider.tsx

```diff
@@ -0,0 +1,72 @@
+/// <reference path="../typings/main.d.ts" />
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import {
+    Component,
+    PropTypes,
+} from 'react';
+
+import {
+    Store,
+} from 'redux';
+
+
+import ApolloClient from 'apollo-client';
+
+declare interface ProviderProps {
+    store?: Store<any>;
+    client: ApolloClient;
+}
+
+export default function composeApolloProvider<T>(Provider: React.ComponentClass<any>): any {
+    return class WrapperProvider extends Component<ProviderProps, any> {
+        /* tslint:disable:member-ordering */
+        static propTypes = {
+            store: PropTypes.shape({
```

Comment:
If we change the underlying implementation of Provider this duplication is cleared